Desktop 4.4.2 release notes

mat007 opened the pull request:
<!--Thanks for your contribution. See [CONTRIBUTING](CONTRIBUTING.md)
    for this project's contribution guidelines. Remove these comments
    as you go.

    Help us merge your changes more quickly by adding details and setting metadata
    (such as labels, milestones, and reviewers) over at the right-hand side.-->

### Proposed changes

Added the Desktop 4.4.2 release notes.

jeanlaurent commented on desktop/mac/release-notes/index.md:
This is unclear IMHO.
`Single Sign-On on docker Hub`
will be more direct.

thaJeztah replied:
Perhaps we should link to the documentation for that; https://docs.docker.com/single-sign-on/ (I think)

mat007 replied:
>  Single Sign-On on hub.docker.com

It’s not just SSO, the username/password flow has changed as well from the in-app dialog to a web page.

mat007 replied:
Added the link to the docs!

stephanierifai commented on desktop/mac/release-notes/index.md:
Should we link the CVE here? Or maybe the security docs page? cc @jerae-duffin

mat007 replied:
I added the link to the CVE as we haven’t yet added a section for it to our docs page.
Obviously we can change that later.

stephanierifai commented on desktop/windows/release-notes/index.md:
same as above

stephanierifai commented on desktop/windows/release-notes/index.md:
Should we link to SSO documentation here?

stephanierifai reviewed (commented):
Maybe add links?

usha-mandya commented on desktop/mac/release-notes/index.md:
Can we add a link to the CVE?

mat007 replied:
Yes, good point!

usha-mandya commented on desktop/mac/release-notes/index.md:
```suggestion
  - Single Sign-on: Users with a Docker Business subscription can now configure SSO to authenticate using their identity providers (IdPs) to access Docker. For more information, see [Single Sign-on](../../../single-sign-on/index.md).
```

mat007 replied:
Awesome, thanks!

mat007 commented:
I added links to the CVE and SSO docs.
I also added a «Known issues» section.

I would really love some English read proofing, especially on the Bug fixes and Know issues sections! 😀

thaJeztah commented on desktop/mac/release-notes/index.md:
I'm wondering if for CVE's we should also include a summary of the vulnerability (and possible recommendations) in the changeling itself, so that users don't have to look elsewhere to collect this information.

From the CVE description:

```markdown
Docker Desktop version 4.3.0 and 4.3.1 contained has a bug that may log sensitive
information (access token or password) on the user's machine during login. This
only affects users if they are on Docker Desktop 4.3.0, 4.3.1 and the user has
logged in while on 4.3.0, 4.3.1. Gaining access to this data would require having
access to the user's local files.
```

And possibly recommendations; something like (please check/review - it's just a quick write-up);

```markdown
Docker Desktop 4.3.2 fixes this bug and remove old log files during upgrade.
While the sensitive information only leaked to files local to the user's machine,
we recommend users to verify if this data has not been stored in external location,
such as back-ups, and to rotate their Docker Hub password in case of concerns.
```

mat007 replied:
Should we just copy the CVE description then?

usha-mandya replied:
@mat007 Discussed this with @stephanierifai and suggest that we add the following CVE description? 

"Docker Desktop version 4.3.0 and 4.3.1 has a bug that may log sensitive information (access token or password) on the user's machine during login. This only affects users if they are on Docker Desktop 4.3.0, 4.3.1 and the user has logged in while on 4.3.0, 4.3.1. Gaining access to this data would require having access to the user’s local files."

mat007 replied:
> Could you also add this to the 4.3.2 release notes please?

Good catch!

mat007 replied:
How do we want to format that? Do we still need the bullet list? If not where do we put the link?

mat007 replied:
I’ve kept both, tell me what you think!

usha-mandya replied:
LGTM.Thanks

thaJeztah commented on desktop/windows/release-notes/index.md:
Was there a reason for removing the `target=blank` on these? (I know we try to use those for external links so that people are more "sticky" on the docs website

mat007 replied:
I honestly have no idea what these do, but we’ve never done that in this section before, so I thought it were a wrong copy-paste of the link further down in the description in the «Security» section. 🤷

stephanierifai replied:
I think it opens the link in a new tab. We should put it back in

usha-mandya commented on desktop/mac/release-notes/index.md:
```suggestion
- Added a deprecated option to `settings.json`: `"deprecatedCgroupsv1": true`, which switches the Linux environment back to cgroups v1. This option will be removed in future releases. If your software requires cgroups v1, you must update it to be compatible with cgroups v2.
```

usha-mandya commented on desktop/mac/release-notes/index.md:
```suggestion
- Fixed an issue where putting the machine to Sleep mode after pausing Docker Desktop results in a failure to resume the machine from the Sleep mode. Fixes [for-mac#6058](https://github.com/docker/for-mac/issues/6058).
```

mat007 replied:
The machine wakes up all right, but Docker cannot resume from pause.
What about:
```suggestion
- Fixed an issue where putting the machine to Sleep mode after pausing Docker Desktop results in Docker Desktop not being able to resume from pause after the machines comes out of Sleep mode. Fixes [for-mac#6058](https://github.com/docker/for-mac/issues/6058).
```
?

usha-mandya replied:
Sounds good, Mathieu.  Just one nit: after the **machine** comes out of Sleep...

stephanierifai replied:
LGTM

usha-mandya reviewed (approved):
LGTM
